ext/ITensorsPackageCompilerExt/precompile_itensors.jl

```diff
@@ -1,4 +1,5 @@
 using ITensors.ITensorMPS: MPO, OpSum, dmrg, random_mps, siteinds
+include("$(@__DIR__)/../ITensorsVectorInterfaceExt/ITensorsVectorInterfaceExt.jl")
```

Comment:
This doesn't look like a good idea to do. Does it work if you load `VectorInterface.jl` to trigger the extension to load, i.e. add the line:
```julia
using VectorInterface: VectorInterface
```
It seems weird that we would have to do that though.

Reply:
I see in https://github.com/ITensor/ITensors.jl/issues/1548#issuecomment-2422883017 you said you already tried that and it didn't work.

The thing is that `VectorInterface.jl` is a dependency of `KrylovKit.jl`, and `KrylovKit.jl` is a dependency of `ITensors.jl`, so `ITensorsVectorInterfaceExt` should get loaded automatically. So there is something strange happening with the order that the packages/extensions get loaded, I suspect it is a bug in Julia and/or Pkg.

Reply:
Something you could try is adding `VectorInterface.jl` as a dependency of `ITensors.jl` directly and see if that forces `ITensorsVectorInterfaceExt` to get loaded. It's implicitly a dependency anyway, as I described, so it shouldn't be a problem adding it as a dependency, though it seems like a bug that we would have to do that.

One reason I made `ITensorsVectorInterfaceExt` is because when we move `ITensors.ITensorMPS` to `ITensorMPS.jl`, `KrylovKit.jl` and `VectorInterface.jl` will no longer be dependencies of `ITensors.jl`, so that extension was anticipating that code reorganization. But if it is causing problems that we can't circumvent right now we could move `ITensorsVectorInterfaceExt` into `ITensors.jl` and turn `VectorInterface.jl` into a dependency.